src/core/datastore/sync/sync-state-manager.js

```diff
@@ -123,8 +124,10 @@ export class SyncStateManager {
   _getCollectionFilter(collection) {
     const result = new Query();
     result.equalTo('collection', collection)
-      .or()
-      .equalTo('collection', stripTagFromCollectionName(collection));
+    if (collectionHasTag(collection)) {
+      result.or()
+        .equalTo('collection', stripTagFromCollectionName(collection));
```

Comment:
Can you explain how this `equalTo` clause would ever match? It seems like it would never match a collection name.

Reply:
The variable `collection`'s value can be either "books" or "books.my-tag". Before the redesign, the value of any sync item's `.collection` property used to be the untagged collection. After the redesign, it's the tagged collection (if it has a tag). So for sync items created before the redesign and still in the queue after an update to the redesigned version, the $or clause must be added - for backwards compatibility. After some time, this can be removed, the assumption being that whatever needed to be synced, was already synced and any new sync items only have the "new" way of setting the value (with the tag).

As to how it would match, it would add an $or clause, if the value of `collection` has a tag. So two cases:

- `collection === 'books'` - no $or statement, just `{ collection: 'books' }`
- `collection === 'books.my-tag'` - $or statement, resulting filter: `{ $or: [{ collection: 'books' }, { collection: 'books.my-tag' }] }`

Am I missing something in the `Query` API? Cause I had a hard time figuring out when to chain calls and when not to. But it appears to work as expected. Which part do you think would fail?

Reply:
It might be that I don't understand what is returned from `stripTagFromCollectionName(collection)`. I am thinking the query filter would be: `{ $or: [{ collection: 'books' }, { collection: 'my-tag' }] }`.

Reply:
Ok this makes sense now to me.